prov/net/src/xnet_progress.c

```diff
@@ -914,10 +852,12 @@ void xnet_progress_rx(struct xnet_ep *ep)
 			ret = ep->cur_rx.handler(ep);
 		}
 
-	} while (!ret && ofi_bsock_readable(&ep->bsock));
+		if (OFI_SOCK_TRY_SND_RCV_AGAIN(-ret))
```

Comment:
Should this check be expanded to include OFI_EINPROGRESS_URING?  It seems that OFI_EINPROGRESS_URING is equivalent to EAGAIN.